integration/spark-datasource/src/main/scala/org/apache/spark/sql/carbondata/execution/datasources/SparkCarbonFileFormat.scala

```diff
@@ -154,8 +154,11 @@ class SparkCarbonFileFormat extends FileFormat
           path
         }
         context.getConfiguration.set("carbon.outputformat.writepath", updatedPath)
+        // "jobid"+"x"+"taskid", task retry should have same task number
         context.getConfiguration.set("carbon.outputformat.taskno",
-          UUID.randomUUID().toString.replace("-", ""))
+          context.getTaskAttemptID.getJobID.getJtIdentifier +
+          context.getTaskAttemptID.getJobID.getId
+          + 'x' + context.getTaskAttemptID.getTaskID.getId)
```

Comment:
Why using `x` in between? Can we follow the same naming convention as carbondata table?

Reply:
@jackylk 
a. 'x' is used to separate jobid and task id.
b. we cannot use logic from carbondata table as taskid, in this case it will be just split index. And this id will be serialized and send to executor. So, during task retry it will use same id. However in case of file format, it was using UUID for each task.

Reply:
@ravipesala : please confirm and share your opinion.

Reply:
@ajantha-bhat can we use underscore as separator??

Reply:
@kumarvishal09 : current file name has many underscore, If some parsing logic depends on underscore, it will have impact. Hence using 'x'

Reply:
@ajantha-bhat then it’s better to use some other characters
Like #,-

Reply:
old file name already use "-", "_"  to separate each parts. 
taskNo is a single part, no need to use "-", "_"  to separate it. 
So we use 'x' to contact  jobId and taskId to a single part of file name.